src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -345,23 +346,9 @@ private void SetUserDetails(K8SConfiguration k8SConfig, Context activeContext)
                             }
                         case "gcp":
                             {
+                                // config
                                 var config = userDetails.UserCredentials.AuthProvider.Config;
-                                const string keyExpire = "expiry";
-                                if (config.ContainsKey(keyExpire))
-                                {
-                                    if (DateTimeOffset.TryParse(config[keyExpire]
-                                                              , out DateTimeOffset expires))
-                                    {
-                                        if (DateTimeOffset.Compare(expires
-                                                                 , DateTimeOffset.Now)
-                                         <= 0)
-                                        {
-                                            throw new KubeConfigException("Refresh not supported.");
-                                        }
-                                    }
-                                }
-
-                                AccessToken = config["access-token"];
+                                TokenProvider = new GcpTokenProvider(config["cmd-path"]);
```

Comment:
I don't know the expected details of a GCP kubeconfig, but is this field present in all configs?

Are the GCP kubeconfig contents documented anywhere?

Reply:
I don't believe google "documents" it as it automatically configures context for you, but it is the way they do it. To be safe I can make this behavior apply only when the specified field is present

Reply:
Yeah, I think I prefer that.

Reply:
I found where this is coming from and in case of "gcp" it will always be present as it's hardcoded behavior of official go client. https://github.com/kubernetes/client-go/blob/master/plugin/pkg/client/auth/gcp/gcp.go

Imo no reason to do the extra check

Reply:
Sounds good, can you add a pointer to this "documentation" in a comment

e.g. 

``// docs: https://github.com/kubernetes/client-go/blob/master/plugin/pkg/client/auth/gcp/gcp.go#L58`